### PR TITLE
feat: Added get user endpoint

### PR DESCRIPTION
**Pull request by marveldo**
## Description  : This update adds a new endpoint [GET] That gets a user with his id
 ## How Has This Been Tested?

 1. **Unit Tests**:
    
* Implemented unit tests for the login endpoint to get the access Token
* I made sure it checks whether the user is authenticated
* I made it give the appropriate response on Success
 ##Screenshot of test
Screenshot 2024-07-29 122924
Screenshot 2024-07-29 123133

Screenshot 2024-07-29 123236
Screenshot 2024-07-29 131115

 ## Related Issue
[Github issue](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/533)
 ## Types of Changes
 
  * [ ]    Bug fix (non-breaking change which fixes an issue)
  * [x]    New feature (non-breaking change which adds functionality)
  * [ ]    Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

* [x]    My code follows the code style of this project.
* [x]    My change requires a change to the documentation.
* [x]    I have updated the documentation accordingly.
* [x]    I have read the CONTRIBUTING document.
* [x]    I have added tests to cover my changes.
* [x]    All new and existing tests passed.

 ## How to Run and Test the Project
### Prerequisites
 Ensure you have the following installed on your machine:

* Python 3.10
* PostgreSQL
* Git
* Virtual env (optional but recommended)

### Setup Instructions

1. **Clone the Repository**
 ```shell
 git clone <repository_url>
cd <repository_directory>
 ```
2. **Create and Activate a Virtual Environment (optional but recommended)**
```shell
python -m venv .venv
source .venv/bin/activate  # On Windows use `.venv\Scripts\activate`
```
3. **Install Dependencies**
```shell
pip install -r requirements.txt 
```
4. **Configure Environment Variables**
Create a `.env` file in the root directory and add the following environment variables:
 
PYTHON_ENV=dev
 
 
DB_TYPE=postgresql
DB_NAME=postgres
DB_USER=postgres
DB_PASSWORD=Winner174
DB_HOST="localhost"
DB_PORT=5432
MYSQL_DRIVER=
DB_URL=postgresql://postgres:Winner174@localhost:5432/postgres
SECRET_KEY = ""
ALGORITHM = HS256
ACCESS_TOKEN_EXPIRE_MINUTES = 10
JWT_REFRESH_EXPIRY=5
APP_URL=

MAIL_USERNAME=""
MAIL_PASSWORD=""
MAIL_FROM=""
MAIL_PORT=465
MAIL_SERVER="smtp.gmail.com"
 ```
5. **Run Database Migrations**
```shell
alembic upgrade head
 ```
 6. **Start the FastAPI Server**
 ```shell
 python main.py
```
   
The server will start at `http://127.0.0.1:7001`.

### Testing Instructions
 
1. **Run Unit Tests with Pytest**
```shell
pytest
``` 
This will run all the test cases and provide a report of the test results.

2. **Test the POST superadmin endpoint**
  
* Open Postman and create a new GET request to `http://127.0.0.1:7001/api/v1/users/user_id.
 * Send the request.
* You should receive a response indicating  that a user has been created and the details of the created user:
```json
  {
 {
  ```

 ```
     {
 "status": 200,
    "message": "User retrieved successfully",
    "user": {
        "id": "066a7873-2373-7d3b-8000-b73f0f688ec4",
        "name": "Marvelous Uboh",
        "email": "utibesolomon6@gmail.com"
    }
 }
 } 
```
 ```
    ``
 ### Notes
 
* Ensure your PostgreSQL database is running and accessible with the credentials provided in the `.env` file.
* If you encounter any issues, please check the logs for detailed error messages and verify that all dependencies are correctly installed.
```

### CODE REVIEW

**Review by Goketech — DISMISSED** [review]

File: api/v1/routes/user.py [hunk 1]
```diff
@@ -98,6 +102,25 @@ async def change_password(
...
     )
 
+@user.get(path="/{user_id}", status_code=status.HTTP_200_OK)
+def get_user(user_id : str,
+             current_user : Annotated[User , Depends(user_service.get_current_user)],
+             db : Session = Depends(get_db)
+             ):
+    user = user_service.fetch(db=db , id=user_id)
+
+    return {
```
Comment (joboy-dev) [inline api/v1/routes/user.py:112]: return the right response. There is a success_response function. Please use that

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

File: api/v1/routes/user.py [hunk 2]
```diff
@@ -98,6 +102,25 @@ async def change_password(
...
+        'status' : status.HTTP_200_OK,
+        'message': 'User retrieved successfully',
+        'user' : UserResponse(id=user.id , name=f'{user.first_name} {user.last_name}' , email=user.email)
```
[the remark this answers is not in the document]
Comment (marveldo) [reply api/v1/routes/user.py]: @joboy-dev  changes have been made

File: api/v1/routes/user.py (hunk 1 above)
Comment (marveldo) [reply api/v1/routes/user.py:112]: @joboy-dev  the success response doesnt follow the format of the issue it uses data and the issue that other teams are using is user

**Review by joboy-dev — APPROVED** [review]

**Comment by joboy-dev** [comment]
@Goketech

**Review by Goketech — APPROVED** [review]

**Review by joboy-dev — APPROVED** [review]

**Pull request merged**